libpgmodeler_ui/src/mainwindow.cpp

```diff
@@ -524,12 +524,33 @@ void MainWindow::restoreTemporaryModels(void)
 
 void MainWindow::showRightWidgetsBar(void)
 {
-	right_wgt_bar->setVisible(objects_btn->isChecked() || operations_btn->isChecked());
-}
+	int i=0;
+	bool show=false;
+	//This is currently the only way to enumerate widgets inside a layout...
+	//See https://stackoverflow.com/a/27225570/7359123
+	while(horizontalLayout_4->itemAt(i))
```

Comment:
What do you mean by "enumerate"? Are you trying to toggle the visibility of children widgets in a layout?

Reply:
Widgets are children of widgets, not layouts. So to find the buttons in a layout this is the only way I found.
Currently this function will display the bottom widget if one of the two buttons are checked.
I make it expandable so plugins can add an arbitrary number of dock widgets.
You made me realize there is an error : it should not check for the layer_btn.